<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.1//EN"
  "http://www.w3.org/TR/xhtml11/DTD/xhtml11.dtd">
<html xmlns="http://www.w3.org/1999/xhtml" xml:lang="en">
<head>
<meta name="generator" content="jemdoc, see http://jemdoc.jaboc.net/" />
<meta http-equiv="Content-Type" content="text/html;charset=UTF-8" />
<link rel="stylesheet" href="jemdoc.css" type="text/css" />
<title><a href="realignment.html" target=&ldquo;blank&rdquo;>Realignment Methods</a></title>
<!-- MathJax -->
<script src='https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/latest.js?config=TeX-MML-AM_CHTML' async>
</script>
<script type="text/x-mathjax-config">
MathJax.Hub.Config({
	  TeX: { equationNumbers: { autoNumber: "AMS" } }
});
</script>
<!-- End MathJax -->
</head>
<body>
<table summary="Table for page layout." id="tlayout">
<tr valign="top">
<td id="layout-menu">
<div class="menu-category">序列比对</div>
<div class="menu-item"><a href="index.html">首页</a></div>
<div class="menu-item"><a href="psa.html">双序列比对</a></div>
<div class="menu-item"><a href="msa.html">多序列比对</a></div>
<div class="menu-item"><a href="measure.html">评分标准</a></div>
<div class="menu-item"><a href="realignment.html" class="current">Realignment</a></div>
<div class="menu-item"><a href="datasets.html">测试数据集</a></div>
<div class="menu-item"><a href="simu_data.html">生成模拟数据</a></div>
<div class="menu-item"><a href="tools.html">实用工具</a></div>
<div class="menu-item"><a href="modules.html">实用开发模块</a></div>
<div class="menu-item"><a href="papers.html">发表论文</a></div>
<div class="menu-item"><a href="https://github.com/malabz" target="blank">Github</a></div>
</td>
<td id="layout-content">
<div id="toptitle">
<h1><a href="realignment.html" target=&ldquo;blank&rdquo;>Realignment Methods</a></h1>
</div>
<h2>Realigners based on horizontal partitioning</h2>
<p>Split an alignment into groups of whole sequences, which are then merged together by realigning between groups, possibly using each group’s induced subalignment.
</p>
<h3><a href="https://dl.acm.org/doi/pdf/10.1145/267521.267524" target=&ldquo;blank&rdquo;>1. ReAligner (1997)</a></h3>
<p>In each iteration has a traversal of all sequence:
</p>
<ol>
<li><p>Pick a sequence
</p>
</li>
<li><p>Realign the sequence with the profile constructed with the remaining sequences.
</p>
</li>
<li><p>If the new alignment's quality is improved, the new one will replace the old one.
</p>
</li>
</ol>
<p>Until the alignment score saturates to a stable value.
</p>
<ul>
<li><p>Test data: DNA/RNA 
</p>
</li>
<li><p><a href="ftp://ftp.cs.arizona.edu/realigner" target=&ldquo;blank&rdquo;>The source code link is unavailable.</a>
</p>
</li>
</ul>
<h3><a href="https://academic.oup.com/bioinformatics/article/21/8/1408/249176" target=&ldquo;blank&rdquo;>2. Remove First method (2005)</a></h3>
<p>In each iteration:
</p>
<ol>
<li><p>Randomly pick a sequence.
</p>
</li>
<li><p>Realign the sequence with the profile constructed with the remaining sequences.
</p>
</li>
<li><p>Update the alignment if the accuracy improved.
</p>
</li>
</ol>
<p>Until the score converges or the reach the limit of iterations.
</p>
<ul>
<li><p>Test data: Protein (HOMSTRAD)
</p>
</li>
<li><p><a href="http://www.bioinf.ucd.ie/people/iain/iteration.html" target=&ldquo;blank&rdquo;>Source code</a>
</p>
</li>
</ul>
<h3><a href="https://ieeexplore.ieee.org/abstract/document/1498013" target=&ldquo;blank&rdquo;>3. TreeRefiner (2005)</a></h3>
<p>This realigner is based on three-dimensional alignment rather than using an iterative approach.
</p>
<ul>
<li><p>Test data: Nucleotide sequences that were generated by the Rose program
</p>
</li>
<li><p><a href="http://treerefiner.stanford.edu/index.html" target=&ldquo;blank&rdquo;>TreeRefiner's homepage</a>
</p>
</li>
</ul>
<h3><a href="https://academic.oup.com/nar/article/34/9/2598/2401672" target=&ldquo;blank&rdquo;>4. REFINER (2006)</a></h3>
<p>This realigner is similar to RF method but aims to realign the sequences with the family block models representing conserved sequence/structure regions.
</p>
<p>In each iteration:
</p>
<ol>
<li><p>Randomly pick a sequence.
</p>
</li>
<li><p>Realign the sequence with the PSSM constructed with the remaining sequences.
</p>
</li>
<li><p>Update the alignment.
</p>
</li>
</ol>
<p>Until the alignment score saturates to a stable value or until the iteration cycle terminates.
</p>
<ul>
<li><p>Test data: Protein (<a href="http://www.ncbi.nlm.nih.gov/Structure/cdd/cdd.shtml" target=&ldquo;blank&rdquo;>CDD</a> and PFAM)
</p>
</li>
<li><p><a href="ftp://ftp.ncbi.nih.gov/pub/REFINER" target=&ldquo;blank&rdquo;>FTP server</a>
</p>
</li>
</ul>
<h3><a href="https://link.springer.com/article/10.1186/1471-2105-15-265" target=&ldquo;blank&rdquo;>5. ReformAlign (2014)</a></h3>
<p>In each iteration:
</p>
<ol>
<li><p>Build a profile from the initial alignment.
</p>
</li>
<li><p>All sequences realign with the profile to obtain the reformed alignment. (There has a profile fine-tuning model.)
</p>
</li>
<li><p>The alignment with better quality will be the input of next iteration.
</p>
</li>
</ol>
<p>Until the alignment between two successive runs remains unchanged or a predefined maximum number of iterations is reached.
</p>
<ul>
<li><p>Test data: DNA/RNA (BRAliBase 2.1 RNA alignment database and DNA SMART database)
</p>
</li>
<li><p><a href="http://evol.bio.lmu.de/_statgen/software/reformalign/" target=&ldquo;blank&rdquo;>ReformAlign's homepage</a>
</p>
</li>
</ul>
<h2><a href="realigner_ver.html" target=&ldquo;blank&rdquo;>Realigners based on vertical partitioning</a></h2>
<h2><a href="realigner_both.html" target=&ldquo;blank&rdquo;>Realigners based on vertical and horizontal partitioning</a></h2>
<h2>Realignment in MSA tools</h2>
<h3><a href="realign_msa_ver.html" target=&ldquo;blank&rdquo;>Vertical-oriented realignment in MSA tools</a></h3>
<h3><a href="realign_msa_hor.html" target=&ldquo;blank&rdquo;>Horizontal-oriented realignment in MSA tools</a></h3>
</td>
</tr>
</table>
</body>
</html>
